rfcs/0012-mtls.md

```diff
@@ -0,0 +1,106 @@
+# RFC Template
+
+- Feature Name: `mtls`
+- Start Date: 2024-11-27
+- RFC PR: [Kuadrant/architecture#110](https://github.com/Kuadrant/architecture/pull/110)
+- Issue tracking: [Kuadrant/kuadrant-operator#1049](https://github.com/Kuadrant/kuadrant-operator/issues/1049)
+
+# Summary
+[summary]: #summary
+
+MTLS support for communication between the Wasm plugin in the gateway pod, and Authorino & Limitador.
+
+# Motivation
+[motivation]: #motivation
+
+For security reasons so that all traffic is encrypted, and certificates validated.
+This aligns with a 'zero trust' architecture.
+
+# Guide-level explanation
+[guide-level-explanation]: #guide-level-explanation
+
+Update the Kuadrant CR to have a new configuration option `mtls` for users to configure if mtls should be enabled for either authorino or limitador.
+The reason for having separate flags for either component is to allow flexibility where the resource overhead of tls may be considered too high when enabled for a specific component e.g. for checking limits on each request in limitador.
+The new config in the Kuadrant CR would look like this:
+
+```yaml
+apiVersion: Kuadrant.io/v1alpha1
+kind: kuadrant
+spec:
+  mtls:
+    authorino: true
+    limitador: true
+```
+
+When enabled for a specific component, the following changes are performed by the kuadrant-operator and/or component operators. This may require trickling down configuration to other operator resources, like the Authorino or Limitador resources:
+
+## Enabling the Istio sidecar proxy in that component
```

Comment:
We should probably add a section for Envoy Gateway as well.

Reply:
I don't know what to add for that.
The Istio approach takes advantage of Istio sidecar capabilities to enable mTLS in the kuadrant pods.
If Istio isn't there, would that be some other solution outside of Envoy Gatway completely?